src/sentry/utils/circuit_breaker.py

```diff
@@ -15,6 +16,42 @@ class CircuitBreakerPassthrough(TypedDict, total=True):
     window: int
 
 
+class CircuitBreakerTripped(Exception):
+    pass
+
+
+class CircuitBreakerConfig(TypedDict, total=False):
+    # The number of consecutive failures within a given window required to trigger the circuit breaker
+    error_limit: int
+    # The window of time in which those errors must happen
+    error_limit_window: int
+    # Allow a configurable subset of function calls to bypass the circuit breaker, for the purposes
+    # of determining when the service is healthy again and requests to it can resume.
+    allow_passthrough: bool
+    # The number of function calls allowed to bypass the circuit breaker every
+    # `passthrough_interval` seconds
+    passthrough_attempts_per_interval: int
+    # The window of time, in seconds, during which to allow `passthrough_attempts_per_interval`
+    # calls to bypass the circuit breaker
+    passthrough_interval: int
+
+
+CIRCUIT_BREAKER_DEFAULTS = CircuitBreakerConfig(
+    error_limit=DEFAULT_ERROR_LIMIT,
+    error_limit_window=3600,  # 1 hour
+    allow_passthrough=False,
+    # TODO: No idea what a reasonable value here is
+    passthrough_interval=15,  # in sec
```

Comment:
seems reasonable to me, the severity limit is 1 per 10 seconds 

```suggestion
    passthrough_interval=15,  # 15 seconds
```